### PR TITLE
Fix things

### PR DESCRIPTION
**Pull request by Atry**
This PR fixes a lot of issues preventing us from submitting a job to AKS:

- Fixed devenv-built OCI images and `libnvidia-container` compatibility issue by cherry-picking  https://github.com/cachix/devenv/pull/873
- Set the customized python package to `pkgs` passed to `poetry2nix`, so that dependents of Python also get rebuilt agained the customized python. Without the changes dependent packages of Python will conflicts with the customized python.
- Added a `maxLayers` option to build multi-layer images by cherry-picking https://github.com/cachix/devenv/pull/872
- Updated the example project to select spot nodes and GPU nodes when submitting a job to AKS

### CODE REVIEW

File: flake-modules/devenv-python-with-libstdc++.nix [hunk 1]
```diff
@@ -9,18 +9,20 @@ topLevel@{ flake-parts-lib, lib, inputs, ... }: {
...
-                (self: super: {
-                  env = super.env // {
-                    # Link libstdc++ to python interpreter so that packages in manylinux ABI can find it out-of-the-box without LD_LIBRARY_PATH
-                    # TODO: Add more libraries here when encountering an ImportError
-                    NIX_LDFLAGS = "--no-as-needed -lstdc++ --as-needed ${super.env.NIX_LDFLAGS}";
-                  };
-                });
-            })
+        ml-ops.common.pythonPackage = {
+          pipe = [
```
Comment (fmmoret) [inline flake-modules/devenv-python-with-libstdc++.nix:13]: what change are you making here? Are pipe and override attributes intrinsic to all nix objects? or are they just names you're using

Comment (fmmoret) [reply flake-modules/devenv-python-with-libstdc++.nix:13]: nvm. I see that they are just constructs that you've made and are manually handling / passing to pipe etc. etc.

**Review by fmmoret — APPROVED** [review]

File: flake-modules/devenv-python-with-libstdc++.nix (hunk 1 above)
Comment (Atry) [reply flake-modules/devenv-python-with-libstdc++.nix:13]: `lib.trivial.pipe` is a nixpkgs function, and `ml-ops.common.pythonPackage.pipe` is an option of the arguments passed to `lib.trivial.pipe`. The `pipe` option introduced by this PR is more general than previous `override` and `overrideAttrs` options.

Comment (Atry) [reply flake-modules/devenv-python-with-libstdc++.nix:13]: The `pipe` option is defined here: https://github.com/Preemo-Inc/nix-ml-ops/pull/14/files/8155c73aad64219963723e6ea9b6dc8748cab958#diff-bc203345847ac84c75ffb360b81bdabaf45b7c96f68b3d10818d92dcdc6885c3R19

**Pull request merged**